src/web/arr/trove/chart.arr

```diff
@@ -75,13 +86,150 @@ fun to-table3(xs :: List<Any>, ys :: List<Any>, zs :: List<Any>) -> TableIntern:
   map3({(x, y, z): [raw-array: x, y, z]}, xs, ys, zs) ^ builtins.raw-array-from-list
 end
 
+fun list-to-table2<A>(table :: List<List<A>>) -> RawArray<RawArray<A>>:
+  builtins.raw-array-from-list(table.map(builtins.raw-array-from-list))
+end
+
+fun table2-to-list<A>(table :: RawArray<RawArray<A>>) -> List<List<A>>:
+  raw-array-to-list(table).map(raw-array-to-list)
+end
+
 fun get-vs-from-img(s :: String, raw-img :: IM.Image) -> VS.ValueSkeleton:
   I.color(190, 190, 190, 0.75)
     ^ IM.text-font(s, 72, _, "", "modern", "normal", "bold", false)
     ^ IM.overlay-align("center", "bottom", _, raw-img)
     ^ VS.vs-value
 end
 
+fun table-sorter<A,B>(
+    t :: TableIntern, 
+    value-getter :: (RawArray -> A), 
+    scorer :: (A -> B), 
+    cmp :: (B, B -> Boolean), 
+    eq :: (B, B -> Boolean)): 
+  doc: ```
+       General Data Table Sorting Function:
+       Value-getter grabs the Column of the Data table you want to use to sort
+       Scorer Modifies the values in that Column to what you want to sort-by
+       ```
+  list-of-rows = t ^ raw-array-to-list
+
+  scored-values = 
+    map(
+      {(row): {row; row ^ value-getter ^ scorer}}, 
+      list-of-rows)
+
+  sorted-by-score = 
+    scored-values.sort-by(
+      {(row-score, oth-row-score): cmp(row-score.{1}, oth-row-score.{1})}, 
+      {(row-score, oth-row-score): eq(row-score.{1}, oth-row-score.{1})})
+
+  sorted-rows = map({(row-score): row-score.{0}}, sorted-by-score)
+  sorted-rows ^ builtins.raw-array-from-list
+end
+
+fun num-to-scientific(base :: Number) -> (Number -> SciNumber) block: 
+  doc: ```
+       Produces a function that takes a number and turns it into it's scientific representation. 
+       Calculates the resulting Coeff, Exponent where number = coeff * base ^ Exponent.
+       Currently only works with bases > 1.
+       ```
+  when base <= 1: 
+    raise("Num-to-scientific: Only defined on bases > 1")
+  end
+  
+  fun recur(s :: SciNumber): 
+    doc: ``` 
+         Takes the current Coeff, Exponent and divides/multiplies by base to move closer to 
+         the actual scientific representation.
+         ```
+    cases (SciNumber) s: 
+      | sci-notation(c, e, b) => 
+        pos-c = num-abs(c)
+        ask: 
+          | (pos-c > 0) and (pos-c < 1) then: recur(sci-notation(c * b, e - 1, b))
+          | (pos-c == 0) or ((pos-c >= 1) and (pos-c < b)) then: sci-notation(c, e, b)
+          | otherwise: recur(sci-notation(c / b, e + 1, b))
+        end
+    end
+  end
+  
+  {(n): recur(sci-notation(n, 0, base))}
```

Comment:
Replace these lines with the more direct
```
  fun convert(n :: Number):
    if num-is-rational(n) and (n == 0): sci-notation(0, 0, base)
    else:
      pow = num-floor(num-log(num-abs(n)) / num-log(base))
      c = n / num-expt(base, pow)
      sci-notation(c, pow, base)
    end
  end
  convert
```

Reply:
Got it, replacing the function now!